iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,359 @@
+            interfacelistOverflowDetected = true;
+            EXPECT_THAT(error, Eq(iox::Error::kPORT_POOL__INTERFACELIST_OVERFLOW));
+        });
+
+    for (auto i = 0u; i < iox::MAX_INTERFACE_NUMBER; ++i)
+    {
+        auto interfacePort = m_runtime->getMiddlewareInterface(iox::capro::Interfaces::INTERNAL);
+        ASSERT_NE(nullptr, interfacePort);
+    }
+
+    EXPECT_FALSE(interfacelistOverflowDetected);
+
+    auto interfacePort = m_runtime->getMiddlewareInterface(iox::capro::Interfaces::INTERNAL);
+
+    EXPECT_EQ(nullptr, interfacePort);
+    EXPECT_TRUE(interfacelistOverflowDetected);
+}
+
+
+TEST_F(PoshRuntime_test, SendRequestToRouDiValidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << m_runtimeName
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_runnableName;
+
+    const auto successfullySent = m_runtime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
+
+    EXPECT_TRUE(m_receiveBuffer.isValid());
+    EXPECT_TRUE(successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, SendRequestToRouDiInvalidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << m_runtimeName
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_invalidRunnableName;
+
+    const auto successfullySent = m_runtime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
+
+    EXPECT_FALSE(successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSenderIsSuccessful)
+{
+    const auto senderPort = m_runtime->getMiddlewareSender(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    ASSERT_THAT(senderPort, Ne(nullptr));
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), senderPort->m_serviceDescription);
+    EXPECT_EQ(22u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSenderDefaultArgs)
+{
+    const auto senderPort = m_runtime->getMiddlewareSender(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSenderSenderlistOverflow)
+{
+    auto senderlistOverflowDetected{false};
+
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&senderlistOverflowDetected](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            if (error == iox::Error::kPORT_POOL__SENDERLIST_OVERFLOW)
+            {
+                senderlistOverflowDetected = true;
+            }
+        });
+
+    ///@note 5 sender ports are alloted for internal services of Roudi.
+    /// hence getServiceRegistryChangeCounter() is used
+    auto serviceCounter = m_runtime->getServiceRegistryChangeCounter();
+    auto availableSenderPort = serviceCounter->load();
```

Comment:
It is a bit unfortunate to derive the number of used ports from this. This is an API limitation of the runtime. Maybe the runtime could be extended with way to query this information.